settings.py

```diff
@@ -61,9 +60,14 @@
 # Enable or disable downloader middlewares
 # See https://docs.scrapy.org/en/latest/topics/downloader-middleware.html
 
+# DOWNLOADER_MIDDLEWARES = {
+#         'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware': 110,
+#         'scraping.middlewares.CustomProxyMiddleware': 543,
+# }
+CustomProxyMiddleware = True
 DOWNLOADER_MIDDLEWARES = {
-        'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware': 110,
-        'scraping.middlewares.ScrapingDownloaderMiddleware': 543,
+    'scraping.middlewares.CustomProxyMiddleware': 350,
+    'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware': 400,
```

Comment:
Ye middleware kaha se ai ha ?

Reply:
ya old wala middleware phala kaha tha jo ma na commet kar deya. pir new wala creat kar deya ha

Reply:
means it was in default setting ?

Reply:
yes

Reply:
By default ye middleware nai hoti settings me.